doc/command_guidelines.md

```diff
@@ -6,13 +6,27 @@ Guidelines marked (*) only apply to command modules, not extensions.
 
 If in doubt, ask!
 
+[1. General Patterns](#general-patterns)
+
+[2. Command Naming and Behavior](#command-naming-and-behavior)
+
+[3. Argument Naming Conventions](#argument-naming-conventions)
+
+[4. Standard Command Types](#standard-command-types)
+
+[5. Non-standard Commands](#non-standard-commands)
+
+[6. Commands with Complex Types](#commands-with-complex-types)
+
+[7. Network Rule Commands](#network-rule-commands)
+
+[8. Coding Practices](#coding-practices)
+
 ## General Patterns
 
 - Be consistent with POSIX tools (support piping, work with grep, awk, jq, etc.)
 - Support tab completion for parameter names and values (e.g. resource names)
-- All commands, command group and arguments must have descriptions
-- You must provide command examples for non-trivial commands
```

Comment:
The linter ensures help for all things. With Aladdin-based examples, it is actually preferable for authors NOT to write examples.